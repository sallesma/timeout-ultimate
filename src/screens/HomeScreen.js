import React from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View, Pressable, Image } from 'react-native';
import { Button } from 'react-native-elements';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { SafeAreaView } from 'react-native-safe-area-context';

import theme from '../utils/theme.js';
import questions from '../../data/questions';
import logo from '../../assets/icon.png';
import I18n from '../utils/i18n';

export default (props) => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.logoContainer}>
        <Image style={styles.logo} source={logo} />
      </View>
      <Text style={styles.title}>{I18n.t('homeScreen.appName')}</Text>
      <Text style={styles.welcomeText}>
        <Text style={styles.emphasis}>{questions.length}</Text>
        {I18n.t('homeScreen.subtitle')}
      </Text>
      <View style={styles.buttonsArea}>
        <Button
          containerStyle={styles.button}
          titleStyle={styles.mainButtonText}
          title={I18n.t('homeScreen.menu.play')}
          onPress={() => props.navigation.navigate('OptionsScreen')}
        />
        <Button
          containerStyle={styles.button}
          title={I18n.t('homeScreen.menu.history')}
          onPress={() => props.navigation.navigate('HistoryScreen')}
        />
        <Button
          containerStyle={styles.button}
          title={I18n.t('homeScreen.menu.rules')}
          onPress={() => props.navigation.navigate('RulesScreen')}
        />
      </View>
      <View style={styles.infoArea}>
        <Pressable onPress={() => props.navigation.navigate('AboutScreen')}>
          <MaterialCommunityIcons name="information-outline" style={styles.icon} />
        </Pressable>
      </View>
      <StatusBar style="auto" />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 16,
    justifyContent: 'space-evenly',
  },
  logoContainer: {
    alignItems: 'center',
    height: 100,
    marginVertical: 8,
  },
  logo: {
    flex: 1,
    height: 100,
    width: 100,
    borderRadius: 50, // round
  },
  title: {
    color: theme.MAIN_COLOR,
    fontSize: theme.FONT_SIZE_XXL,
    fontWeight: 'bold',
    textAlign: 'center',
    margin: 16,
  },
  welcomeText: {
    fontSize: theme.FONT_SIZE_L,
    marginBottom: 32,
    textAlign: 'center',
  },
  emphasis: {
    color: theme.MAIN_COLOR,
    fontSize: theme.FONT_SIZE_XL,
    fontWeight: 'bold',
  },
  buttonsArea: {
    marginHorizontal: 16,
  },
  button: {
    marginBottom: 16,
    borderRadius: 8,
  },
  mainButtonText: {
    fontSize: theme.FONT_SIZE_XL,
  },
  infoArea: {
    marginHorizontal: 8,
    alignItems: 'flex-end',
  },
  icon: {
    fontSize: theme.FONT_SIZE_XXL,
    padding: 8,
  },
});
