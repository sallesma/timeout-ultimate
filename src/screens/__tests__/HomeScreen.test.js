import React from 'react';
import { render } from '@testing-library/react-native';

import HomeScreen from '../HomeScreen';

describe('<HomeScreen />', () => {
  it('renders correctly', async () => {
    const { toJSON } = render(<HomeScreen />);

    expect(toJSON()).toMatchSnapshot();
  });
});
