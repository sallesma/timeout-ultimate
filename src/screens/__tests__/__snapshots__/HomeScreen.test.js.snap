// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-evenly",
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 100,
        "marginVertical": 8,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 50,
          "flex": 1,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#2a9d9d",
        "fontSize": 32,
        "fontWeight": "bold",
        "margin": 16,
        "textAlign": "center",
      }
    }
  >
    Timeout Ultimate
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 32,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2a9d9d",
          "fontSize": 26,
          "fontWeight": "bold",
        }
      }
    >
      269
    </Text>
     questions t'attendent pour apprendre les règles de l'ultimate !
  </Text>
  <View
    style={
      Object {
        "marginHorizontal": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          Object {
            "borderRadius": 8,
            "marginBottom": 16,
          },
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 26,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Jouer
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          Object {
            "borderRadius": 8,
            "marginBottom": 16,
          },
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Historique
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          Object {
            "borderRadius": 8,
            "marginBottom": 16,
          },
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Règlement
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "marginHorizontal": 8,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text />
    </View>
  </View>
</RNCSafeAreaView>
`;
